test/controllers/v1/PayeRoutingSpec.scala

```diff
@@ -0,0 +1,141 @@
+package controllers.v1
+
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ FreeSpec, Matchers }
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import support.matchers.HttpServerErrorStatusCodeMatcher.aServerError
+
+/*
+ * We are relying on the router to perform argument validation for us (via regex constraints).
+ * This spec tests that the router is configured correctly.
+ *
+ * Because we are relying on the router to validate arguments, and the simple routes file configuration does not
+ * compose, each and every route must be tested individually.  In addition, some of the regex are non-trivial,
+ * and as the saying goes:
+ *
+ *     "Some people, when confronted with a problem, think 'I know, I'll use regular expressions.'
+ *     Now they have two problems."
+ *
+ * For example, in order to fully test the period regex, we test that each and every possible month is considered
+ * valid.  These are downsides of this "router based validation" approach ...
+ */
+class PayeRoutingSpec extends FreeSpec with Matchers with GuiceOneAppPerSuite with ScalaFutures {
+
+  /*
+   * When valid arguments are routed to the retrieve... actions, an attempt will be made to make a call to
+   * sbr-control-api.  However, we do not prime a response in this spec, as we are only concerned with routing.
+   * We therefore override the HTTP timeout configuration to minimise the time this spec waits on a connection that
+   * we know cannot be established.
+   */
+  override def fakeApplication(): Application = new GuiceApplicationBuilder().configure(Map("play.ws.timeout.connection" -> "50")).build()
+
+  private trait Fixture {
+    val ValidPayeRef = "125H7A71620"
+    val ValidPeriod = "201804"
+  }
+
+  "A request to retrieve a PAYE unit by PAYE reference and period" - {
+    "is rejected when" - {
+      "the VAT reference" - {
+        "has fewer than four characters" in new Fixture {
```

Comment:
Can we double check PAYE minimum length is 4 - I recall there was confusion with the minimum being either 4 or 6

Reply:
This has been confirmed by the product owner in relation to historic data that we have reveived.